src/definitions/behaviors/form.js

```diff
@@ -1388,7 +1390,22 @@
                                     var invalidFields = module.validate.rule(field, rule, true) || [];
                                     if (invalidFields.length > 0) {
                                         module.debug('Field is invalid', identifier, rule.type);
-                                        fieldErrors.push(module.get.prompt(rule, field));
+                                        var fieldError = module.get.prompt(rule, field);
+                                        if (!settings.inline) {
+                                            if (
+                                                // Always allow the first error prompt for new field identifiers
+                                                (!(identifier in formErrorsTracker)
+                                                // Also allow multiple error prompts per field identifier but make sure each prompt is unique
+                                                || !formErrorsTracker[identifier].indexOf(fieldError))
```

Comment:
This won't work as `indexOf` does not return a boolean value. infact if 0 is returned the fieldError is actually found at position 0
As you want to check if fieldError does not exists in the whole array, check for -1 then
```suggestion
                                                || formErrorsTracker[identifier].indexOf(fieldError) === -1)
```

Reply:
Thank you for pointing out my careless mistake, I did not see any problems while testing it out. I do agree that a stricter comparison is safer.

Applied the suggestion under commit [ae3e0ec](https://github.com/fomantic/Fomantic-UI/pull/3067/commits/ae3e0ecfed6706e2bc7bd66e6ab3ac2987c9a7f9).